pennylane/ops/qubit/observables.py

```diff
@@ -488,10 +558,166 @@ def compute_diagonalizing_gates(
 
         **Example**
 
-        >>> qml.Projector.compute_diagonalizing_gates([0, 1, 0, 0], wires=[0, 1])
+        >>> _BasisStateProjector.compute_diagonalizing_gates([0, 1, 0, 0], wires=[0, 1])
         []
         """
         return []
 
-    def pow(self, z):
-        return [copy(self)] if (isinstance(z, int) and z > 0) else super().pow(z)
+
+class _StateVectorProjector(Observable):
+    # The call signature should be the same as Projector.__new__ for the positional
+    # arguments, but with free key word arguments.
+    def __init__(self, state, wires, do_queue=None, id=None):
```

Comment:
not sure where I stand, but do you think we should validate that the norm of the state is 1? we do it for `QubitStateVector` as an example

Reply:
At the beginning we thought about this possibility with @albi3ro, but computing the norm could potentially be expensive for large state vectors. I guess we should define the use case we want to cover (small circuits or big simulations?) and adjust the trade-off between efficiency and safety